Update documentation of KNX sync_state and expected DPTs

farmio opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
- Explain all functions of the `sync_state` attribute for sensor and binary_sensor.
- Add expected DPT to all group addresses.
- Removed deprecated `significant_bit` attribute in binary_sensor.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [x] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

The changes documented here are already released with home-assistant 0.115.0

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

farmio committed:
updates for xknx 0.13

farmio committed:
textlint

farmio commented on source/_integrations/binary_sensor.knx.markdown:
```diff
@@ -8,21 +8,13 @@ ha_iot_class: Local Push
 ha_domain: knx
 ---
 
-<div class='note'>
-  
-The `knx` integration must be configured correctly to use this integration, see [KNX Integration](/integrations/knx).
-
-</div>
```
I've removed these notes as the platforms can't be configured without first configuring knx: now since https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md

marvin-w commented on source/_integrations/knx.markdown:
```diff
@@ -111,7 +111,7 @@ knx:
 {% configuration %}
 local_ip:
-  description: The local IP address of interface (which should be used for multicasting).
+  description: The local IP address of interface that shall send multicast packets.
```
```suggestion
  description: The local IP address of the interface that shall send multicast packets.
```

marvin-w reviewed (dismissed):
Only one small minor change that could make sense. Also, we still need to add the docs to one big single documentation.

farmio committed:
review typo

marvin-w reviewed (dismissed):
Looks fine.

farmio commented:
These updates are for HA 0.115 which will be released today. Do I have to change the target branch to `current` when this is not merged before release? 
Or shall I open a new PR?

farmio commented:
I have changed the base branch and rebased from `current`. There were 3 commits in this PR that are not mine... I reverted them. No idea why this happens.

frenck commented:
You cannot go in and just revert those. You need to git rebase this PR and clean up.

frenck reviewed (changes requested):
Please cleanup the PR by rebasing and removing all unrelated commits.

frenck reviewed (approved):
Thanks, @farmio 👍
